Based on original Debian patch, updated from 6.14 to 6.20 and 6.23

silently ignore bad colour variables, for defaults have unsupported ones

https://bugs.gentoo.org/120792

--- a/tw.color.c
+++ b/tw.color.c
@@ -381,16 +381,14 @@
 			if ((Char)variables[i].variable[0] == (v[0] & CHAR) &&
 			    (Char)variables[i].variable[1] == (v[1] & CHAR))
 			    break;
+		    v += 3;
 		    if (i < nvariables) {
-			v += 3;
 			getstring(&c, &v, &variables[i].color, ':');
 			if (i == VSym)
 			    color_as_referent = strcasecmp(
 				variables[VSym].color.s, "target") == 0;
-			continue;
 		    }
-		    else
-			stderror(ERR_BADCOLORVAR, v[0], v[1]);
+		    continue;
 		}
 		break;
 	    }
